logstash-core/src/main/java/org/logstash/instrument/metrics/counter/LazyDelegatingCounter.java

```diff
@@ -0,0 +1,79 @@
+package org.logstash.instrument.metrics.counter;
+
+import org.logstash.instrument.metrics.AbstractMetric;
+import org.logstash.instrument.metrics.MetricType;
+
+import java.util.List;
+
+/**
+ * A lazy proxy to a more specific typed {@link CounterMetric}. The metric will only be initialized if an initial value is provided, or once an {@code increment} operation is
+ * called.
+ * <p><strong>Intended only for use with Ruby's duck typing, Java consumers should use the specific typed {@link CounterMetric}</strong></p>
+ */
+public class LazyDelegatingCounter extends AbstractMetric<Long> implements CounterMetric<Long> {
```

Comment:
Given that we only have a single `LongCounter` maybe we can just delete this class and use `LongCounter directly`.